plasmapy/particles/particle_collections.py

```diff
@@ -0,0 +1,296 @@
+
+    @property
+    def half_life(self) -> u.s:
+        """An array of the half-lives of the particles."""
+        return self._get_particle_attribute("half_life", unit=u.s, default=np.nan * u.s)
+
+    def insert(self, index, particle: ParticleLike):
+        """Insert a particle before an index."""
+        # TODO: use particle_input when it works with CustomParticle and ParticleLike
+        if not isinstance(particle, (Particle, CustomParticle)):
+            particle = Particle(particle)
+        self.data.insert(index, particle)
+
+    @property
+    def integer_charge(self) -> np.array:
+        """
+        An array of the quantized charges of the particles, as multiples
+        of the elementary charge.
+        """
+        return np.array(self._get_particle_attribute("integer_charge", default=np.nan))
+
+    @property
+    def mass(self) -> u.kg:
+        """An array of the masses of the particles."""
+        return self._get_particle_attribute("mass", unit=u.kg, default=np.nan * u.J)
+
+    @property
+    def mass_energy(self) -> u.J:
+        """
+        An array of the mass energies of the particles in joules.
+
+        If the particle is an isotope or nuclide, return the mass energy
+        of the nucleus only.
+        """
+        return self._get_particle_attribute(
+            "mass_energy", unit=u.J, default=np.nan * u.J,
+        )
+
+    def sort(self, key: Callable = None, reverse: bool = False):
+        """
+        Sort the `ParticleList` in-place.  For more information, refer to the documentation for `list.sort`.
+        """
+        if key is None:
+            raise TypeError("Unable to sort a ParticleList without a key.")
+        else:
+            self._data.sort(key=key, reverse=reverse)
+
+    @property
+    def symbols(self) -> List[str]:
+        """A list of the symbols of the particles."""
+        return self._get_particle_attribute("symbol")
+
+
+ParticleList.clear.__doc__ = """Remove all items from the `ParticleList`."""
+
+ParticleList.copy.__doc__ = """Return a shallow copy of the `ParticleList`."""
+
+ParticleList.count.__doc__ = """
+Return the number of occurrences of ``value``.  Here, ``value`` may be a
+`Particle`, `CustomParticle`, or `ParticleLike` representation of a particle.
+"""
+
+ParticleList.extend.__doc__ = """
+Extend `ParticleList` by casting `ParticleLike` items from ``iterable``
+into `Particle` or `CustomParticle` instances.
+"""
+
+ParticleList.index.__doc__ = """
+Return first index of a `ParticleLike` value. Raise `ValueError` if
+the value is not present.
+"""
+
+ParticleList.pop.__doc__ = """
+Remove and return item at index (default last).  Raise `IndexError` if
+the `ParticleList` is empty or the index is out of range.
+"""
+
+ParticleList.remove.__doc__ = """
+Remove the first occurrence of a `ParticleLike` item.  Raise
+`ValueError` if the value is not present.
+"""
+
+ParticleList.reverse.__doc__ = """Reverse the `ParticleList` in place."""
```

Comment:
This is to override the docstrings for `collections.UserList` which were either absent or clunky.  The implementations for each of these methods didn't need to be overridden.